Xamarin.Forms.Core/Picker.cs

```diff
@@ -8,63 +9,325 @@ namespace Xamarin.Forms
-				if (eh != null)
-					eh(bindable, EventArgs.Empty);
-			}, coerceValue: CoerceSelectedIndex);
+		public static readonly BindableProperty SelectedIndexProperty =
+			BindableProperty.Create(nameof(SelectedIndex), typeof(int), typeof(Picker), -1, BindingMode.TwoWay,
+									propertyChanged: OnSelectedIndexChanged, coerceValue: CoerceSelectedIndex);
+
+		public static readonly BindableProperty ItemsSourceProperty =
+			BindableProperty.Create(nameof(ItemsSource), typeof(IList), typeof(Picker), default(IList),
+									propertyChanged: OnItemsSourceChanged);
+
+		public static readonly BindableProperty SelectedItemProperty =
+			BindableProperty.Create(nameof(SelectedItem), typeof(object), typeof(Picker), null, BindingMode.TwoWay,
+									propertyChanged: OnSelectedItemChanged);
 
 		readonly Lazy<PlatformConfigurationRegistry<Picker>> _platformConfigurationRegistry;
 
 		public Picker()
 		{
-			Items = new ObservableList<string>();
-			((ObservableList<string>)Items).CollectionChanged += OnItemsCollectionChanged;
+			((INotifyCollectionChanged)Items).CollectionChanged += OnItemsCollectionChanged;
 			_platformConfigurationRegistry = new Lazy<PlatformConfigurationRegistry<Picker>>(() => new PlatformConfigurationRegistry<Picker>(this));
 		}
 
-		public IList<string> Items { get; }
+		public IList<string> Items { get; } = new LockableObservableListWrapper();
+
+		public IList ItemsSource
+		{
+			get { return (IList)GetValue(ItemsSourceProperty); }
+			set { SetValue(ItemsSourceProperty, value); }
+		}
 
 		public int SelectedIndex
 		{
 			get { return (int)GetValue(SelectedIndexProperty); }
 			set { SetValue(SelectedIndexProperty, value); }
 		}
 
-		public Color TextColor
+		public object SelectedItem
 		{
+			get { return GetValue(SelectedItemProperty); }
+			set { SetValue(SelectedItemProperty, value); }
+		}
+
+		public Color TextColor {
 			get { return (Color)GetValue(TextColorProperty); }
 			set { SetValue(TextColorProperty, value); }
 		}
 
-		public string Title
-		{
+		public string Title {
 			get { return (string)GetValue(TitleProperty); }
 			set { SetValue(TitleProperty, value); }
 		}
 
+		BindingBase _itemDisplayBinding;
+		public BindingBase ItemDisplayBinding {
+			get { return _itemDisplayBinding; }
+			set {
+				if (_itemDisplayBinding == value)
+					return;
+
+				OnPropertyChanging();
+				var oldValue = value;
+				_itemDisplayBinding = value;
+				OnItemDisplayBindingChanged(oldValue, _itemDisplayBinding);
+				OnPropertyChanged();
+			}
+		}
+
 		public event EventHandler SelectedIndexChanged;
 
+		static readonly BindableProperty s_displayProperty =
+			BindableProperty.Create("Display", typeof(string), typeof(Picker), default(string));
+
+		string GetDisplayMember(object item)
+		{
+			if (ItemDisplayBinding == null)
+				return item.ToString();
+
+			ItemDisplayBinding.Apply(item, this, s_displayProperty);
+			ItemDisplayBinding.Unapply();
+			return (string)GetValue(s_displayProperty);
```

Comment:
Awesome! Gonna try find me some good resources on `BindingBase` - seems like I overlooked it :)